plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/templates/appengine/AppEngineTemplateUtility.java

```diff
@@ -0,0 +1,66 @@
+package com.google.cloud.tools.eclipse.util.templates.appengine;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Map;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.FileLocator;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import com.google.common.base.Preconditions;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateExceptionHandler;
+
+public class AppEngineTemplateUtility {
+  public static final String APPENGINE_WEB_XML_TEMPLATE = "appengine-web.xml.ftl";
+  public static final String HELLO_APPENGINE_TEMPLATE = "HelloAppEngine.java.ftl";
+  public static final String INDEX_HTML_TEMPLATE = "index.html.ftl";
+  public static final String WEB_XML_TEMPLATE = "web.xml.ftl";
+
+  private static Configuration configuration;
+
+  public static void createFileContent(String outputFileLocation, String templateName, Map<String, String> dataMap)
+      throws CoreException {
+    Preconditions.checkNotNull(outputFileLocation, "output file is null");
+    Preconditions.checkNotNull(templateName, "template name is null");
+    Preconditions.checkNotNull(dataMap, "data map is null");
+
+    try {
+      if (configuration == null) {
+        configuration = createConfiguration();
```

Comment:
Maybe this is called while a workspace is locked, so no issue with the thread-safety?